app/views/handlers/individual_response.py

```diff
@@ -147,6 +151,26 @@ def _publish_fulfilment_request(self, mobile_number=None):
             topic_id, message=fulfilment_request.payload
         )
 
+    def _check_individual_response_count(self):
+        if (
+            self._questionnaire_store.response_metadata.get(
+                "individual_response_count", 0
+            )
+            >= current_app.config["EQ_INDIVIDUAL_RESPONSE_LIMIT"]
+        ):
+            raise IndividualResponseLimitExceeded(
+                "Individual response limit has been exceeded"
+            )
+
+    def _update_individual_response_count(self):
+        response_metadata = self._questionnaire_store.response_metadata
+
+        if response_metadata.get("individual_response_count"):
+            response_metadata["individual_response_count"] += 1
+        else:
+            response_metadata["individual_response_count"] = 1
+        self._questionnaire_store.save()
```

Comment:
Personally, I think all of this is more appropriate as part of the `FulfilmentRequest` class.

Reply:
I had it in there to begin with, but moved it out. I didn't like the extra logic in publish_fulfilment_request. Publish to me feels like something that happens after checks have been done, I contemplated a name change but decided against it. I also felt if you had hit the limit you shouldn't be updating the status (although you could argue its wrong way round anyway) and doing a check on selected option is redundant

Reply:
Not sure about that one, it feels like it should be resolved sooner and not effectively under _publish_fulfilment_request which to me is a method once everything has been checked. As a kicker as soon as I put it in FulfilmentRequest I won't be able to use you test handler as I will lack app_context

Reply:
I'm not against avoiding it to prevent the need for app context in tests tbh. But maybe `_publish_fulfilment_request` could handle the check to see if it is possible to publish before carrying out the action?

Reply:
I agree it should be kept outside of the `FulfilmentRequest` class, @MebinAbraham's suggestion of checking in `_publish_fulfilment_request` makes sense, as that would cover both Post and Mobile options.